packages/protocol/test-sol/governance/network/Governance.t.sol

```diff
@@ -0,0 +1,3698 @@
+pragma solidity ^0.5.13;
+
+import "celo-foundry/Test.sol";
+import "solidity-bytes-utils/contracts/BytesLib.sol";
+import "openzeppelin-solidity/contracts/cryptography/ECDSA.sol";
+
+// Contract to test
+import "../../../contracts/governance/Governance.sol";
+import "../../../contracts/governance/Proposals.sol";
+import "../../../contracts/governance/test/MockLockedGold.sol";
+import "../../../contracts/governance/test/MockValidators.sol";
+import "../../../contracts/governance/test/TestTransactions.sol";
+import "../../../contracts/common/Accounts.sol";
+import "../../../contracts/common/Signatures.sol";
+import "../../../contracts/common/Registry.sol";
+import "../../../contracts/common/FixidityLib.sol";
+
+contract GovernanceForTest is Governance(true) {
+  address[] validatorSet;
+
+  // Minimally override core functions from UsingPrecompiles
+  function numberValidatorsInCurrentSet() public view returns (uint256) {
+    return validatorSet.length;
+  }
+
+  function numberValidatorsInSet(uint256) public view returns (uint256) {
+    return validatorSet.length;
+  }
+
+  function validatorSignerAddressFromCurrentSet(uint256 index) public view returns (address) {
+    return validatorSet[index];
+  }
+
+  // Expose test utilities
+  function addValidator(address validator) external {
+    validatorSet.push(validator);
+  }
+
+  // exposes removeVotesWhenRevokingDelegatedVotes for tests
+  function removeVotesWhenRevokingDelegatedVotesTest(address account, uint256 maxAmountAllowed)
+    public
+  {
+    _removeVotesWhenRevokingDelegatedVotes(account, maxAmountAllowed);
+  }
+
+  function setDeprecatedWeight(address voterAddress, uint256 proposalIndex, uint256 weight)
+    external
+  {
+    Voter storage voter = voters[voterAddress];
+    VoteRecord storage voteRecord = voter.referendumVotes[proposalIndex];
+    voteRecord.deprecated_weight = weight;
+  }
+
+  function getHotfixHash(
+    uint256[] calldata values,
+    address[] calldata destinations,
+    bytes calldata data,
+    uint256[] calldata dataLengths,
+    bytes32 salt
+  ) external pure returns (bytes32) {
+    return keccak256(abi.encode(values, destinations, data, dataLengths, salt));
```

Comment:
Wondering if it's worth adding this function to the Governance contract itself. It has the downside that'll have to audit a contract with not real change, but we could expose that function for the public.

Reply:
But since we're likely auditing Governance because of #10731, it may be worth add it.

Reply:
Added it to the Governance contract, and modified `executeHotfix` to use the `getHotfixHash` function.

That being said, I'm not sure it really adds much to the contract, other than increase gas usage when calling `executeHotfix()`